cpp/src/Ice/Makefile.mk

```diff
@@ -16,18 +16,27 @@ endif
 
 Ice_excludes            = src/Ice/DLLMain.cpp
 Ice[shared]_excludes    = src/Ice/RegisterPluginsInit_min.cpp
-Ice[xcodesdk]_excludes  = src/Ice/RegisterPluginsInit_min.cpp
-Ice[static]_excludes    = src/Ice/RegisterPluginsInit_all.cpp
 
 ifeq ($(os),Linux)
 ifeq ($(shell pkg-config --exists libsystemd 2> /dev/null && echo yes),yes)
 Ice_cppflags                            += -DICE_USE_SYSTEMD $(shell pkg-config --cflags libsystemd)
 endif
 endif
 
-Ice[iphoneos]_excludes                  := $(wildcard src/Ice/CtrlCHandler.cpp $(addprefix $(currentdir)/,Tcp*.cpp Service.cpp))
-Ice[iphoneos]_extra_sources             := $(wildcard $(addprefix $(currentdir)/ios/,*.cpp *.mm))
-Ice[iphonesimulator]_excludes           = $(Ice[iphoneos]_excludes)
-Ice[iphonesimulator]_extra_sources      = $(Ice[iphoneos]_extra_sources)
+ios_extrasources :=  $(wildcard $(addprefix $(currentdir)/ios/,*.cpp *.mm))
+ios_excludes := $(wildcard $(addprefix $(currentdir)/,\
+	CtrlCHandler.cpp \
+	OutputUtil.cpp \
+	RegisterPluginsInit_all.cpp \
+	Service.cpp \
+	SysLoggerI.cpp \
+	SystemdJournalI.cpp \
+	Tcp*.cpp))
```

Comment:
Should we also exclude UDP? WS?

Reply:
I don't think so. We support WS and UDP, we exclude TCP because there is an iOS implementation in  `ios/StreamXxx`